### ISSUE TITLE
Stop creating desktop shortcuts on Itch update

### ISSUE THREAD

**Issue by Deozaan** [itchio/itch#239]
Every time I download an update to Itch, it creates a new shortcut on the desktop.

Please stahp! :)

Windows 10 v1511

**Comment by fasterthanlime** [comment]
A new one? uh, that's unexpected. I guess we could remove old one, create new ones on update

**Comment by Deozaan** [comment]
Sorry, I wasn't clear.

I don't like desktop shortcuts. I delete them.

It doesn't make multiple shortcuts appear on the desktop. It just keeps bringing the shortcut back when updating Itch after it (the shortcut) has been deleted.

I don't want the shortcut on my desktop. One in the start menu is fine, but please stop recreating the desktop shortcut every time itch updates, especially given the frequent updates.

**Comment by fasterthanlime** [comment]
Oh, I see.

So, we call for shortcut creation in https://github.com/itchio/itch/blob/master/app/util/reg.js — whether it is on install or update (cf. https://github.com/itchio/itch/blob/master/app/util/auto-updater/win32.js#L37 )

I suppose we could call it only on install. The reasoning behind calling it all the time was for the protocol handler, earlier versions didn't register it, so I wanted to make sure everyone had it. I suppose we could split that in two methods.

**Comment by smiley** [comment]
You needn't create desktop shortcuts, one in the start menu is enough. Windows calls those out specifically:
![image](https://cloud.githubusercontent.com/assets/354181/12071833/9cd9317e-b0c9-11e5-8407-8c3d1b213e20.png)
_(Windows 7 highlights them in the "All Programs" menu)_

**Comment by fasterthanlime** [comment]
@smiley As far as I can tell, Squirrel.Windows doesn't have the option of "not creating a desktop shortcut". I think it's okay for an app to create a desktop shortcut on initial installation, but if you delete it, it should stay deleted.

@Deozaan looked a bit more into it, and
- We do need to call createShortcut both on install & update — because it needs to update the shortcut on update.
- However, Squirrel.Windows has code to _not_ re-create the shortcut if it's not the initial installation and the shortcut is no longer there: [see relevant code](https://github.com/Squirrel/Squirrel.Windows/blob/11f09efa4fa414823e909b8b5f48d419865474d3/src/Squirrel/UpdateManager.ApplyReleases.cs#L218-L225)

So, it looks like our setup is correct, it's just not doing the right thing? Let me try to reproduce with an upgrade between 0.11.3 and 0.11.4 ..

**Comment by Deozaan** [comment]
I'm a bit puzzled about why the shortcut needs to be updated on update. Is the location or filename of the executable being changed between updates?

Aside from that, I'm still on 0.11.0 so I figured I'd try with an upgrade to the latest version, but I'm getting an error when checking for updates:

![itch error](https://cloud.githubusercontent.com/assets/227734/12128093/62c6d8a0-b3b8-11e5-900d-a1d320a9243a.png)

(The error message either goes away too fast or doesn't seem to provide any way to get more information about it or even easily copy the full text of the error, hence the screenshot.)

**Comment by fasterthanlime** [comment]
> I'm a bit puzzled about why the shortcut needs to be updated on update. Is the location or filename of the executable being changed between updates?

Yes. Take a look into `%APPDATA%/Local/itch` — it's part of the way Squirrel.Windows works, it allows very nice things like update-while-the-app-is-running, and reverting to a non-broken version easily, etc.

> Aside from that, I'm still on 0.11.0 so I figured I'd try with an upgrade to the latest version, but I'm getting an error when checking for updates:

_sigh_ I was pretty sure that wasn't an issue anymore :( going to look into it..

**Comment by fasterthanlime** [comment]
@Deozaan the update problem should be solved now.

(Note to self: to avoid a big headache, don't skip version numbers...)

**Comment by fasterthanlime** [comment]
@Deozaan I'm starting to suspect that same problem for updates.

Let me explain:
- To push a release, I push a git tag
- The release is then built on several nodes, orchestrated by http://ci.itch.ovh
- When the release is built, it's pushed to GitHub (also by jenkins) as _private_, so it doesn't affect updates on OSX & Windows, and, similarly, it's pushed to Bintray as drafts to our deb & rpm repositories (which expire after 7 days if I don't publish them) by hand
- I then download it & test it on various computers I have at home (Mac, Linux, Windows)
- Sometimes I find bug during the QA phase, so I decide the release isn't good enough
- Usually I just deleted the release and pushed the next version number

But then...
- The windows node has a locally cached version of the 'skipped' version
- Squirrel.Windows creates a delta file from the skipped version to the actually-released version
- When trying to upgrade, it tries to download a delta-file from before-skipped to skipped, then from skipped to actually-released - obviously, that fails because the first delta file doesn't exist (that's the error you saw)
- When solving that manually (by removing all traces of skipped-version from the RELEASES file and reuploading it), it still doesn't update correctly because the delta is indeed from skipped-version to actually-released, not from actual-previous-version to actually-released
- Since it can't apply delta files, it downloads the full package and installs it
- And **that** - I think - is why it creates a shortcut again. It doesn't consider it an update, but an install from scratch..

TL;DR from 0.11.3 on it shouldn't be a problem anymore.

**Comment by fasterthanlime** [comment]
Nevermind, installing 0.11.3 then upgrading to 0.11.4 (normally, via delta files - checked with http://fiddler2.com/) does recreate the shortcuts. Hmmm.

**Comment by fasterthanlime** [comment]
Atom used to have a separate issue, we can probably apply their workaround: https://github.com/atom/atom/issues/4665

**Comment by Deozaan** [comment]
> Sometimes I find bug during the QA phase, so I decide the release isn't good enough

Er... Shouldn't QA happen before the release is pushed and version number is bumped?

Regardless, thanks for looking into this, and all the work you're doing for Itch.

**Comment by fasterthanlime** [comment]
> Er... Shouldn't QA happen before the release is pushed and version number is bumped?

For some reason I forgot to mention in my original listing that the released is pushed as _private_ on GitHub which means it doesn't show up to everyone else :)

As for versions, pushing tags is the way the jenkins setup generates production-like builds. What I should've done though, is deleting + force-pushing a new tag when finding a bug in QA, instead of skipping a version.

**Comment by fasterthanlime** [comment]
> Shouldn't QA happen before the release is pushed

Additionally: I do run itch many times on all 3 OSes before I even decide to try making a production-like build, but sometimes issues only show up in production-like builds.

**Comment by Deozaan** [comment]
I just updated from 0.11.4 to 0.12.4 and can confirm no new shortcut created on my desktop. Thanks!

**Comment by fasterthanlime** [comment]
@Deozaan victory! :tada: :balloon: :birthday: :beer:
